Source/WebCore/dom/StyledElement.cpp

```diff
@@ -230,7 +230,9 @@ void StyledElement::styleAttributeChanged(const AtomString& newStyleString, Attr
     if (document().scriptableDocumentParser() && !document().isInDocumentWrite())
         startLineNumber = document().scriptableDocumentParser()->textPosition().m_line;
 
-    if (reason == ModifiedByCloning || document().contentSecurityPolicy()->allowInlineStyle(document().url().string(), startLineNumber, newStyleString.string(), CheckUnsafeHashes::Yes, *this, nonce(), isInUserAgentShadowTree()))
+    if (newStyleString.isNull())
+        ensureMutableInlineStyle().clear();
+    else if (reason == ModifiedByCloning || document().contentSecurityPolicy()->allowInlineStyle(document().url().string(), startLineNumber, newStyleString.string(), CheckUnsafeHashes::Yes, *this, nonce(), isInUserAgentShadowTree()))
```

Comment:
It seems like we should see some kind of behavior change in our testing here. We either need to create a new test to help us avoid making this mistake in the future, or we should modify (or unskip) an existing test that may be marked as expected to fail.

Do you know if there is an existing Web Platform Test case that covers this? If so, we could import it as part of this change.

Reply:
I don't see that WPT has an existing test for this. The test I added should cover it.